cshlatz/cs-spiff
# ProgressBar exercise V1

To the reviewer: [V2](https://github.com/cshlatz/cs-spiff/pull/2)

### Summary
My implementation of a loading bar, per the V1 specs on the Spiff exercise page. Note that the package files have been changed - when I brought this repo down I was having quite a bit of trouble getting sass to run. I've used it pretty regularly in the past so I'm assuming there was some interference. I wiped it and reinstalled it.

### Testing
- Make sure you have the `testing-library` packages I added installed so the custom `useInterval` hook can be tested.
- `cd tests && npm run test` (or `yarn test`)
- Run through the requirements for the V1 Progress Bar and make sure they're all fulfilled. Try out the edge cases with the button while the progress bar is active
- Make sure the progress bar fades away after three seconds.

### Tasks
- [x] Progress bar works as expected
- [x] Make sure it passes linting
- [x] Prop types (done in V2)
